### ISSUE TITLE
RedisGraph crashes multiple times in a row.

### ISSUE THREAD

**Issue by arramos84** [RedisGraph/RedisGraph#1341]
While running redisgraph as a deployment with no replicas in our development environment, we encountered 4 crashes within ~20 minutes.

[crash1.txt](https://github.com/RedisGraph/RedisGraph/files/5229226/crash1.txt)
[crash2.txt](https://github.com/RedisGraph/RedisGraph/files/5229233/crash2.txt)
[crash3.txt](https://github.com/RedisGraph/RedisGraph/files/5229234/crash3.txt)
[crash4.txt](https://github.com/RedisGraph/RedisGraph/files/5229235/crash4.txt)

Thanks!

@DvirDukhan , I am with the Asserts team.

**Comment by manoja1** [comment]
This crash has been going on for some time now. We are using the latest image. The redisgraph container crashes and then crashloops and restarts couple of times and then startup and keeps humming for few days. This is a dev instance, so irregular request pattern.

**Comment by DvirDukhan** [comment]
can you validate graph version is `2.0.20`  (latest) by 
`MODULE LIST`

**Comment by arramos84** [comment]
```
>> module list
1) 1) "name"
   2) "graph"
   3) "ver"
   4) (integer) 20020
```

It looks like this query causes the crash:

```
match (n:Pod) where n.namespace='asserts' return n union match (s:Service) where n.namespace = 'asserts' return n
```
thanks!

**Comment by DvirDukhan** [comment]
@arramos84 
notice that the second part of the query is matching a node for the variable `s` but filters on the variable `n` 
```
match (s:Service) where n.namespace = 'asserts' return n
```
This trigger an assertion that is still present in the `2.0` branch, but is fixed on `master` on #1319  and we will backport this soon to `2.0.21`

**Comment by arramos84** [comment]
@DvirDukhan , we are just pulling the docker image from `redisgraph:latest`, so will either live with it for now until it's updated or pull from `redisgraph:edge`.

Thanks!
